### ISSUE TITLE
assess Debugger.gd

### ISSUE THREAD

**Issue by TheDuriel** [moonwards1/Moonwards-Virtual-Moon#124]
what even does this file do? theres a lot of misc functions in there doing all sorts of rather complex operations.

**Comment by yaelatletl** [comment]
Mostly adressed by #142, but may be worth giving it a second look.

**Comment by TheDuriel** [comment]
what happens if we just delete it and try to run the project?

**Comment by yaelatletl** [comment]
Currently it's an autoload, so it will probably crash

**Comment by TheDuriel** [comment]
Note: File is now called Debugger.gd

**Comment by TheDuriel** [comment]
after removing the singleton:

```

func set_fps_limit(value : int) -> void:
	Options.set("dev", value, "3FPSlimit_value")
	Debugger.set_3fps(t_FPSLimit.pressed, value)
	
```
in OptionsUi.gd

this seems to be the only error raised by the removal. the game still opens into the main and options menu

this _also_ appears to have fixed the lag spike when trying to open and close the options menu.

and loading into a local game is much faster now...

**Comment by yaelatletl** [comment]
Changing it to an instanceable class broke the performance monitor and other options in the Dev tab

**Comment by TheDuriel** [comment]
Yeah that is not the right approach.

First, vital live code, like fps limits, need to be moved out of the class. The debugger can then be turned into a monitoring class, provided information of the UI readouts.

But more importantly, it currently causes significant performance issues. Those need to be resolved first.

**Comment by TheDuriel** [comment]
closed by #145
